### ISSUE TITLE
Repo: Migrate to Nx executors for more commands

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6142]
### Suggestion

We've previously migrated to Nx for linting (#6038). Let's do that for more commands.

Looking at testing in particular, we might consider using the [Nrwl Jest plugin](https://nx.dev/packages/jest)!

```shell
nx g @nrwl/jest:jest-project --project=<project-name>
```

```json
"test": {
  "executor": "@nrwl/jest:jest",
  "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
  "options": {
    "jestConfig": "packages/typescript-estree/jest.config.ts",
    "passWithNoTests": true
  }
}
```

However, we should note that this might not get us much benefit. We're already running Jest and the Nx runner is going to do that in a slightly different way. We'll see.

**Comment by JoshuaKGoldberg** [comment]
Yeah, I played with this a bit locally and it seemed pretty equivalent. FYI @JamesHenry I'll close this out.

But if we ever see benefits from commands to migrate to Nx, then I'd be up for switching over. 🙂
